zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java

```diff
@@ -670,6 +676,100 @@ public void shouldApplyOutputMappings() {
         .contains(tuple("adHocResult", "[1,2]", processInstanceKey));
   }
 
+  @Test
+  public void shouldSetAdHocSubProcessElementsVariable() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlClasspathResource("/processes/ad-hoc-sub-process-elements.bpmn")
+        .deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable("activateElements", List.of("Task_A"))
+            .create();
+
+    final var adHocSubProcessKey =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId(AD_HOC_SUB_PROCESS_ELEMENT_ID)
+            .getFirst()
+            .getKey();
+
+    // then
+    assertThat(
+            RecordingExporter.variableRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .filter(
+                    v ->
+                        v.getIntent() == VariableIntent.CREATED
+                            && v.getValue().getName().equals(AD_HOC_SUB_PROCESS_ELEMENTS_VARIABLE))
+                .limit(1))
+        .first()
+        .describedAs(
+            "Variable adHocSubProcessElements should be created as local variable in sub-process scope")
+        .satisfies(
+            variableRecord -> {
+              assertThat(variableRecord.getValue()).hasScopeKey(adHocSubProcessKey);
```

Comment:
Is this enough to assert this has been created in the local scope of the ad-hoc sub-process?